views/containers/GameLoader.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Game from '../components/Game';
+import LoadingScreen from '../components/LoadingScreen';
+import * as BrowserActions from '../actions/Browser';
+
+class GameLoaderComponent extends React.Component {
+  constructor() {
+    super();
+    this.didLoad = false;
+  }
+
+  componentDidMount() {
+    setTimeout(()=>{
+      this.didLoad=true;
+      this.forceUpdate()//fake server load
+    },4000);
+
+    this.props.onDidMount();
+  }
+  componentWillMount() {
+    this.props.onWillMount();
+  }
+
+  componentWillUnmount(){
+    this.props.onWillUnmount();
+  }
+
+  render() {
+    if (!this.didLoad)
+      return <LoadingScreen size={this.props.size}/>;;
+    return <Game size={this.props.size}/>;
+  }
+}
+
+GameLoaderComponent.propTypes = {
+  onDidMount: React.PropTypes.func.isRequired,
+  onWillMount: React.PropTypes.func.isRequired,
+  onWillUnmount:React.PropTypes.func.isRequired,
+  size: React.PropTypes.shape({
+    width: React.PropTypes.number,
+    height: React.PropTypes.number,
+  })
+}
+
+const mapDispatchToProps = (dispatch) => {
+  const resize = ()=>{dispatch(BrowserActions.updateViewportSize())}
+  const onDidMount = ()=>{
+    dispatch(BrowserActions.addEventListener('Game', 'resize', resize));
```

Comment:
I insist that is normal for component to dispatch actions, and doing this adds unnecessary levels of layers to finally dispatch actions.